client/web/src/components/exit-node-selector.tsx

```diff
@@ -27,17 +28,42 @@ export default function ExitNodeSelector({
   const api = useAPI()
   const [open, setOpen] = useState<boolean>(false)
   const [selected, setSelected] = useState<ExitNode>(toSelectedExitNode(node))
+  const [pending, setPending] = useState<boolean>(false)
+  const { mutate } = useSWRConfig() // allows for global mutation
   useEffect(() => setSelected(toSelectedExitNode(node)), [node])
+  useEffect(() => {
+    // Delay with a clear timeout to avoid UI flickering between pending
+    // and advertising UI when the node is already approved as an exit node
+    // on the admin side before enabling as an exit node through web client.
+    const timeout = setTimeout(() => {
+      setPending(
+        selected.ID === runAsExitNode.ID &&
+          node.AdvertisingExitNodeApproved === false
+      )
+    }, 500)
+
+    return function () {
+      clearTimeout(timeout)
+    }
+  }, [selected, node])
 
   const handleSelect = useCallback(
     (n: ExitNode) => {
       setOpen(false)
       if (n.ID === selected.ID) {
         return // no update
       }
+      // Eager clear of pending state to avoid UI oddities
+      if (n.ID !== runAsExitNode.ID) {
+        setPending(false)
+      }
       api({ action: "update-exit-node", data: n })
+
+      setTimeout(() => {
+        mutate("/data")
```

Comment:
Nit: add comment

`// refresh data again after short timeout to pick up any pending approval updates`